frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadList/uploadListItem/components/uploadListItemDestination/uploadListItemDestination.component.tsx

```diff
@@ -15,34 +15,123 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
+import React, { useState } from 'react';
+import ChevronIcon from '@assets/icons/chevron.svg';
+import ClearIcon from '@assets/icons/clear_circle.svg';
+import MuiAutocomplete, { createFilterOptions } from '@material-ui/lab/Autocomplete';
+import { DestinationOption } from '@/v5/store/containers/containers.types';
+import { ContainersHooksSelectors } from '@/v5/services/selectorsHooks/containersSelectors.hooks';
+import { useFormContext } from 'react-hook-form';
+import { ErrorTooltip } from '@controls/errorTooltip';
+import { TextInput } from './uploadListItemDestination.styles';
+import { NewContainer } from './options/newContainer';
+import { ExistingContainer } from './options/existingContainer';
 
-import { Tooltip } from '@material-ui/core';
-import ErrorIcon from '@assets/icons/error_circle.svg';
-import { Input } from './uploadListItemDestination.styles';
+interface IUploadListItemDestination {
+	onChange: (option) => void;
+	errorMessage: string;
+	disabled?: boolean;
+	className?: string;
+}
 
-type IUploadListItemDestination = {
-	isSelected: boolean;
-	errorMessage?: string;
+const emptyOption = {
+	containerId: '',
+	containerName: '',
+	latestRevision: '',
 };
+const filter = createFilterOptions<DestinationOption>();
 
-export const UploadListItemDestination = ({
+export const UploadListItemDestination: React.FC<IUploadListItemDestination> = ({
 	errorMessage,
-	isSelected,
+	disabled = false,
+	className,
+	onChange,
 	...props
-}: IUploadListItemDestination): JSX.Element => (
-	<Input
-		$selectedrow={isSelected}
-		error={!!errorMessage}
-		InputProps={{
-			startAdornment: !!errorMessage && (
-				<Tooltip title={errorMessage} placement="bottom-start">
-					<span>
-						<ErrorIcon />
-					</span>
-				</Tooltip>
-			),
-		}}
-		{...props}
-	/>
-);
+}) => {
+	const [value, setValue] = useState<DestinationOption>(emptyOption);
+	const [newOrExisting, setNewOrExisting] = useState('unset');
+	const containers = ContainersHooksSelectors.selectContainers();
+
+	const [containersInUse, setContainersInUse] = useState([]);
+	const { getValues } = useFormContext();
+	const forceUpdate = React.useCallback(() => {
+		const values = getValues().uploads;
+		setContainersInUse(values.map((val) => val.containerName));
+	}, []);
+
+	return (
+		<MuiAutocomplete
+			popupIcon={<ChevronIcon />}
+			closeIcon={<ClearIcon />}
+			openText=""
+			closeText=""
+			clearText=""
+			handleHomeEndKeys
```

Comment:
Lets put all of these in the theme, props section. Checkout MuiCheckbox to see how the icons are set.